src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -213,6 +225,16 @@ public SCMHeadOrigin getOrigin() {
         return origin == null ? SCMHeadOrigin.DEFAULT : origin;
     }
 
+    /**
+     * Get pull request labels
+     * @return Labels
+     * @since TODO
+     */
+    @NonNull
+    public Set<String> getLabels() {
```

Comment:
Nope Nope Nope! The labels will form part of the equality check and thus you will force a rebuild every time you change the labels